platform/darwin/src/NSExpression+MGLAdditions.mm

```diff
@@ -1046,6 +1055,10 @@ - (id)mgl_jsonExpressionObject {
             
         case NSFunctionExpressionType: {
             NSString *function = self.function;
+
+            BOOL hasCollectionProperty = !( ! [self.arguments.firstObject isKindOfClass: [NSExpression class]] || self.arguments.firstObject.expressionType == NSConstantValueExpressionType || self.arguments.firstObject.expressionType == NSKeyPathExpressionType
```

Comment:
A collection expression has a `NSAggregateExpressionType` have you consider simplifying this check using the later?